WalletWasabi/Models/SmartLabel.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace WalletWasabi.Models
+{
+	public class SmartLabel : IEquatable<SmartLabel>
+	{
+		public static SmartLabel Empty { get; } = new SmartLabel();
+		public static char[] Separators { get; } = new[] { ',', ':' };
+		public IEnumerable<string> Labels { get; }
+		public bool IsEmpty { get; }
+
+		private string LabelString { get; }
+
+		public SmartLabel(params string[] labels) : this(labels as IEnumerable<string>)
+		{
+		}
+
+		public SmartLabel(IEnumerable<string> labels)
+		{
+			labels = labels ?? Enumerable.Empty<string>();
+			Labels = labels
+				   .SelectMany(x => x?.Split(Separators, StringSplitOptions.RemoveEmptyEntries) ?? new string[0])
+				   .Select(x => x.Trim())
+				   .Where(x => x != "")
+				   .Distinct(StringComparer.OrdinalIgnoreCase)
+				   .OrderBy(x => x)
```

Comment:
This is just a comment and not a suggestion to change anything now:
It would be good to consider the user's culture when we order the labels because it looks incorrect with other than English language.